Chaincase.UI/Services/StackService.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Chaincase.UI.Services
+{
+	public class StackService
+	{
+		public event Action StackStateChanged;
+		private Stack<Func<Task>> _stackedStates = new Stack<Func<Task>>();
+		public bool AnyStackState => _stackedStates.Any();
+
+		public Task InvokeStackState()
+		{
+			if (AnyStackState)
+			{
+				return _stackedStates.Pop().Invoke().ContinueWith(task => StackStateChanged?.Invoke());
+			}
+
+			return Task.CompletedTask;
```

Comment:
this is kinda confusing to me, it pops the top and invokes the task at the new top I guess?

Reply:
So it looks a bit funky in this iteration but what happens is that the stacked item is actually a func that should take you to that stack level.